### PR TITLE
[SDK]: Grab functionality added to actor

### PR DESCRIPTION
**Pull request by tombuMS**
Grab is being added to the actors within the sdk as a property on the actor called `grabbable`.  By setting this field, a developer can enable/disable grab on an actor.  Additionally there is now a method on actor that allows a developer to set a handler to be called when a grab begins and/or ends.  This is the new method is `onGrab(grabState: 'begin' | 'end', handler: ActionHandler)`.

Unity runtime reference: [PR#58](https://github.com/Microsoft/mixed-reality-extension-unity/pull/58)

### CODE REVIEW

File: packages/sdk/src/types/runtime/actor.ts [hunk 1]
```diff
@@ -98,8 +99,11 @@ export class Actor implements ActorLike, Patchable<ActorLike> {
     private _text: Text;
     private _attachment: Attachment;
     private _lookAt: LookAt;
+    private _grabbable = false;
     // tslint:enable:variable-name
 
+    private grab = new DiscreteAction();
```
Comment (eanders-MS) [inline packages/sdk/src/types/runtime/actor.ts]: >grab [](start = 12, length = 4)

Most objects won't be grabbable. Can `grab` be lazily allocated?

Comment (tombuMS) [reply packages/sdk/src/types/runtime/actor.ts]: I don't see why not, but we don't actually do that anywhere else in the behavior system at the moment.

Comment (tombuMS) [reply packages/sdk/src/types/runtime/actor.ts]: I don't see why not, but we don't actually do that anywhere else in the behavior system at the moment.

Comment (eanders-MS) [reply packages/sdk/src/types/runtime/actor.ts]: Curious: Would `grab` become a component once we have that system in place?

---
In reply to: [266113938](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/260#discussion_r266113938) [](ancestors = 266113938)

Comment (tombuMS) [reply packages/sdk/src/types/runtime/actor.ts]: I don't know.  My initial gut reaction would be possibly, but it is more like a property of the actor.  Not sure at this moment.  I can see arguments for either way.

File: packages/functional-tests/src/tests/grab-test.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            this.state = 1;
+            this.cycleState();
+        });
+        this.model.onGrab('end', _ => {
+            this.state = 2;
+            this.cycleState();
+        });
+
+        // Create two grabbable cubes that can be played with at will.  Position left
+        // anr right of the monkey.
```
Comment (eanders-MS) [inline packages/functional-tests/src/tests/grab-test.ts]: >anr [](start = 11, length = 3)

anr -> and

**Pull request merged**